src/app/shared/model/base.model.ts

```diff
@@ -126,13 +128,34 @@ export abstract class BaseModel {
         });
         this.relatedReplays[relName] = relValue;
       }
-      return this.relatedReplays[relName];
+      return this.relatedReplays[relName].first();
     } else {
-      let allRelated = this.RELATIONS.map(r => this.loadRelated(r, force).first());
+      let allRelated = this.RELATIONS.map(r => this.loadRelated(r, force));
       return Observable.forkJoin(allRelated).map(relateds => null);
     }
   }
 
+  swapRelated(): Observable<boolean[]> {
```

Comment:
This is the black magic.  If a model defines a `swapNew()` method (couldn't figure out a very typescript-y way of checking for this, other than `if (model['swapNew'])`...

After saving itself, the parent model will force reloading that relation with `this.loadRelated('things', true)`.  Then it will swap out the previous `this.things` with the newly loaded values.  And allow the old model to copy/set its properties to the new model.

This is all to allow the user to set distribution/episode form data ahead of time, even though those records are actually created on the server side.